libecole/src/scip/scimpl.cpp

```diff
@@ -137,5 +203,48 @@ auto ReverseBranchrule::scip_execlp(SCIP* scip, SCIP_BRANCHRULE* /*branchrule*/,
 	return action_func(scip, result);
 }
 
+}  // namespace
+
+/*******************************
+ *  Definition of ReverseHeur  *
+ *******************************/
+
+namespace {
+
+scip::ReverseHeur::ReverseHeur(
+	SCIP* scip,
+	std::weak_ptr<utility::Controller::Executor> weak_executor_,
+	int trials_per_node_,
+	int depth_freq,
+	int depth_start,
+	int depth_stop) :
+	::scip::ObjHeur(
+		scip,
+		scip::ReverseHeur::name,
+		"Primal heuristic that waits for another thread to provide a primal solution.",
+		'e',
+		scip::ReverseHeur::max_priority,
+		depth_freq,
+		depth_start,
+		depth_stop,
+		SCIP_HEURTIMING_AFTERNODE,
+		false),
+	weak_executor(std::move(weak_executor_)),
+	trials_per_node(trials_per_node_) {}
+
+auto ReverseHeur::scip_exec(
+	SCIP* scip,
+	SCIP_HEUR* /*heur*/,
+	SCIP_HEURTIMING /*heurtiming*/,
+	SCIP_Bool /*nodeinfeasible*/,
+	SCIP_RESULT* result) -> SCIP_RETCODE {
+	if (weak_executor.expired() or trials_per_node == 0) {
```

Comment:
```suggestion
	if (weak_executor.expired() || (trials_per_node == 0)) {
```

Reply:
I don't mind that. As a rule, do we avoid the `or` and `and` keywords ?